### ISSUE TITLE
[BUG] When copying contents to a new item, the item is suffixed with ",,,,,,,VSCODE.RESTFS"

### ISSUE THREAD

**Issue by JeffTeter** [mvextensions/mvbasic#133]
Copying the contents of one record to a new empty record, the data from the original does appear in the new record but it also has the string ",,,,,,,VSCODE.RESTFS" at the end of the record. This string does not display in the VSCode editor panel but is seen when looking at the item in UniVerse.

**To Reproduce**
1. Open a text item in the VSCode editor.
2. Select the complete item by doing <CTRL>+A.
3. Copy the contents using <CTRL>+C.
4. Create a new file in the VSCode editor.
5. Paste in the original record using <CTRL>+V.
6. Save the new record.
7. In UniVerse, edit or display the contents of the new record using CT or similar.
8. The string mentioned will be present in the last row of the record.

**Expected behavior**
Copying a record should only result in the contents of the record being copied, without additional data.

**Screenshots**

    >SSELECT MLISTS LIKE [20201117]
      
    10 record(s) selected to SELECT list #0.
    >>CT MLISTS
    You have an active SELECT list.
    
    IRS1098_20201117.CODES
    0001 SYSSCH-MNTR*WNB.CL-875
    0002 SYSSCH-MNTR*WSB.CL-875
    0003 SYSSCH-PROG*WNB.CL-875
    0004 SYSSCH-PROG*WSB.CL-875
    0005 SYSSCH-SCHED*WNB.CL-875
    0006 SYSSCH-SCHED*WSB.CL-875 
    0007 EXPJOB*EXTRACT_CL-875,,,,,,,VSCODE.RESTFS

**Versions of applicable software**
Detail on the versions of any and all pieces of software that are part of your environment, such as:
  * Host OS - Windows 10 Enterprise, v10.0.15299 Build 16299
  * Remote OS - Redhat v7.9
  * Extension - MV Basic v2.1.1
  * AccuTerm v8.0.1020 (with VSCode)
  * VSCode Version: 1.51.1 (user setup)
  * VSCode Commit: e5a624b788d92b8d34d1392e4c4d9789406efe8f
  * Date: 2020-11-10T23:34:32.027Z
  * Electron: 9.3.3
  * Chrome: 83.0.4103.122
  * Node.js: 12.14.1
  * V8: 8.3.110.13-electron.0

**Comment by itsxallwater** [comment]
@JeffTeter can you confirm your AccuTerm host programs version for us?

CC @pschellenbach

**Comment by JeffTeter** [comment]
![image](https://user-images.githubusercontent.com/13050714/100152057-b948d780-2e67-11eb-930e-ec9e86eb8e9e.png)

I am also running the updated 'FTBP FTFSI' component to allow the excluded/included files.

**Comment by JeffTeter** [comment]
FTBP FTFSI v8.0.4
FTBP VSCODE v8.0.3
FTBP WED v8.0.1

Which others would be helpful?

**Comment by itsxallwater** [comment]
Run `FTABOUT` and you should have output like this:

```
jsh ACCUTERM ~ -->FTABOUT
AccuTerm Host Program Details

  FTBP version: 8.000.007
  Host platform: JB
  GUIBP: installed
  OBJBP: installed
  SUIBP: installed
  ACCUTERM account name: ACCUTERM
  DEMOBP: installed
  UIEX: installed
  Sample programs: installed
  Sample data files: installed

File Transfer Configuration
  Default transfer mode: Text
  Host item size limit: Unlimited
  Preserve file extension: Yes
  Overwrite existing items: FT=Yes FTD=Yes FTPICK=Yes
  Text file field delimiter: Tab
  Character set conversion: No
  Source buffer index: 250
  Host requires extra trailing attribute mark: No

Kermit Configuration
  Maximum packet size: 94
  Number of retry attempts: 10
  Error timeout: 8
  Number of padding characters: 0
  End-of-line character: 13
  Prefix characters (CTL, EBP, RLE): # & ~
  Use eighth-bit prefixing: Yes
  Use run-length encoding: No
  Use MCP conversion: Yes

Server Configuration (FTSERVER, WED, GED)
  Host server name:
  Allow READ/READNEXT/RELEASE: Yes
  Allow WRITE/ADD/DELETE: Yes
  Allow CONV: Yes
  Allow EXECUTE/CALL: Yes

AccuTerm Client Details
  Product class: Unknown
  Client platform: Unknown
  Release: 0 (0`)
  Serial number: 0
  License: Invalid
  Registered user:
  Capability codes:
  Service codes:
```

**Comment by JeffTeter** [comment]
Sorry to take so long getting this... Out of pocket over the holidays...

>FTABOUT
AccuTerm Host Program Details

  FTBP version: 8.000.005
  Host platform: UV
  GUIBP: not installed
  OBJBP: not installed
  SUIBP: not installed
  ACCUTERM account name: BLMASTERJJT

File Transfer Configuration
  Default transfer mode: Text
  Host item size limit: Unlimited
  Preserve file extension: Yes
  Overwrite existing items: FT=Yes FTD=Yes FTPICK=Yes
  Text file field delimiter: Tab
  Character set conversion: No
  Source buffer index: 250
  Host requires extra trailing attribute mark: No

Kermit Configuration
  Maximum packet size: 94
  Number of retry attempts: 10
  Error timeout: 8
  Number of padding characters: 0
  End-of-line character: 13
  Prefix characters (CTL, EBP, RLE): # & ~
  Use eighth-bit prefixing: Yes
  Use run-length encoding: No
  Use MCP conversion: Yes

Server Configuration (FTSERVER, WED, GED)
  Host server name:
  Allow READ/READNEXT/RELEASE: Yes
  Allow WRITE/ADD/DELETE: Yes
  Allow CONV: Yes
  Allow EXECUTE/CALL: Yes

AccuTerm Client Details
  Product class: AccuTerm Win32
  Client platform: Win32
  Release: 8.01020 (8.0.1020)
  Serial number: YFCU-NAZK-GLQW-CTK4-9XKJ-YPVC
  License: Organization
  Registered user: jteter@woodforest.com, Woodforest National Bank
  Capability codes: AKFJMPQTVYCDUEILRSOXGH
  Service codes: 0abg

**Comment by itsxallwater** [comment]
No problem! It looks like you're a couple releases behind on the host programs, are you able to update and re-test once you're running `8.000.007`? We've got a callout on this at https://github.com/mvextensions/mvbasic/blob/main/doc/Extension%20Guide.md#52-option-2---accuterm but you can get direct to the download at https://zdownload.blob.core.windows.net/accuterm/Host-Programs/Windows/x86/8.0.10.0/pickbp.exe.

**Comment by pschellenbach** [comment]
Jeff - I tested host programs version 8.0.11 (https://github.com/zumasys/accuterm-host-programs/blob/master/install/pickbp-8.0.11/pickbp.exe) on Universe (older release) and am unable to reproduce the problem.
Thanks, Pete



Peter Schellenbach | Director of Product Development
Zumasys Inc.
818-658-1592 | 866-582-8447
peters@zumasys.com | www.zumasys.com
1050 Calle Amanecer, Suite A | San Clemente, CA 92673

**Comment by itsxallwater** [comment]
Thanks Pete. Good to acknowledge that I am running, and mentioned 8.0.7 whereas the link I posted (from the docs) was to 8.0.10 and Pete has linked an even newer build at 8.0.11, just in case there's any confusion. If memory is serving I'd expect this to address itself with any of those versions but definitely just jump straight to latest!

**Comment by JeffTeter** [comment]
This issue has recurred and I have a bit more of the train of sequences to provide.

1. Each of these files were created by adding a new item to the MLISTS file that is used in our system to hold manifest lists for deployment. These were created as new/empty file.
2. Rows were copied from an existing item (for instance "2020.3.90.36.3.D_WNB!CODES"), pasted into the new file and saved. It appeared to be good on the screen and did not show extra characters. 
3. I went to the TCL level on my server and looked at the item. At this point, I saw that the extra text was saved into the item (but did not show on the VSCode item).
4. I edited the record at TCL and saved the item.
5. I refreshed the folder and reopened the items in question. The item that had been fixed at TCL looked correct but the other items had the extra text.
6. I deleted the extra text from the item "2020.3.90.36.3.D_TRAIN!CUSTOMER" and saved it.
7. I looked at the item at TCL and saw that the extra text was gone.

In conclusion, the issue has to do with copying data into an empty record. What shows on the screen shows only the copied data but what gets recorded to the file has the additional text related to RESTFS.

The attached file include screen shots and the FTABOUT information for AccuTerm.

[VSCode RESTFS issue  on save.docx](https://github.com/mvextensions/mvbasic/files/5667409/VSCode.RESTFS.issue.on.save.docx)

**Comment by pschellenbach** [comment]
Jeff - Thanks for verifying the host programs version. 8.0.11 is the latest version. One other thing to check is the version of atsocksvr.exe that is installed. AccuTerm 8.0.1020 ships with atsocksvr.exe version 1.0.3, but atsocksvr.exe 1.0.4 is available by installing the accuterm-vscode-connector update for AccuTerm 7 users (https://zdownload.blob.core.windows.net/accuterm/Miscellaneous/Windows/x86/1.0.6.0/accuterm-vscode-connector.exe) . This will probably downgrade your FTBP programs, so you'll need to update them again (no need to reinstall on the host). Sorry. A new AccuTerm 8 build will be ready soon which will have everything updated.
Thanks, Pete



Peter Schellenbach | Director of Product Development
Zumasys Inc.
818-658-1592 | 866-582-8447
peters@zumasys.com | www.zumasys.com
1050 Calle Amanecer, Suite A | San Clemente, CA 92673

**Comment by JeffTeter** [comment]
I have tested this again after updating per recommendation from Pete to 'atsocksvr.exe' v1.0.0.4. The process of copying data from one record to a new one seems to be resolved. However, there does still seem to be an issue when copying an item to a new item. That is documented in the attached.

[20201210 Test AccuTerm RESTFS.pdf](https://github.com/mvextensions/mvbasic/files/5675988/20201210.Test.AccuTerm.RESTFS.pdf)

**Comment by pschellenbach** [comment]
Thanks Jeff. I am able to reproduce the problem now. It looks like it affects both the 'rename' and 'copy' REST APIs.


Peter Schellenbach | Director of Product Development
Zumasys Inc.
818-658-1592 | 866-582-8447
peters@zumasys.com | www.zumasys.com
1050 Calle Amanecer, Suite A | San Clemente, CA 92673

**Comment by itsxallwater** [comment]
Live!
